src/otx/core/schedulers/callable.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+"""Scheduler callable to support hyper-parameter optimization (HPO) algorithm."""
+
+from __future__ import annotations
+
+import importlib
+import inspect
+from typing import TYPE_CHECKING, Any
+
+from lightning.pytorch.cli import ReduceLROnPlateau
+from torch import nn
+from torch.optim import Optimizer
+from torch.optim.lr_scheduler import LRScheduler
+from torch.optim.lr_scheduler import ReduceLROnPlateau as TorchReduceLROnPlateau
+
+from otx.core.utils.jsonargparse import ClassType, lazy_instance
+
+if TYPE_CHECKING:
+    from lightning.pytorch.cli import LRSchedulerCallable
+
+
+class SchedulerCallableSupportHPO:
```

Comment:
It's clear that this is a class to support HPO, but in the actual code it appears to be a Callable for Pickable.  Wouldn't the existing `PicklableLRSchedulerCallable` be a better name?

Reply:
> the scheduler callable should also be picklable and allow access on its parameters.
It should satisfy both aspects: 1) picklable and 2) param accessible